metrics/gin_middleware.go

```diff
@@ -4,15 +4,21 @@ import (
 	"github.com/gin-gonic/gin"
 )
 
+var notFoundContextKey = "routeNotFound"
+
 func requestMetricsMiddleware() gin.HandlerFunc {
 	if metrics, enabled := Get(); enabled {
 		return func(c *gin.Context) {
 			tracker := metrics.httpRequests.Track(c.Request.Method, c.Request.URL.Path)
 			tracker.Start()
 			defer func() {
```

Comment:
one thing just came to me, do we really need this defer?
why can't we just put that code at the end of the function?

Reply:
One benefit I can see - of using defer - is: when a next handler throw panic, the defer func with `tracker.End` will be executed.

Reply:
ok, didn't thought about that.